src/rules/incognia-login.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @title Incognia Login Rule
+ * @overview Verify if the device logging in is at a trusted location.
+ * @gallery true
+ * @category marketplace
+ *
+ */
+async function incogniaLoginRule(user, context, callback) {
+  const _ = require('lodash@4.17.19')
+
+  const { IncogniaAPI } = require('@incognia/api@1.0.0')
+  const { Auth0UserUpdateUtilities } = require('@auth0/rule-utilities@0.2.0')
+
+  const { INCOGNIA_CLIENT_ID, INCOGNIA_CLIENT_SECRET } = configuration;
+
+  if (!INCOGNIA_CLIENT_ID || !INCOGNIA_CLIENT_SECRET) {
+    console.log('Missing required configuration. Skipping.');
+    return callback(null, user, context);
+  }
+
+  // For this rule to be used, please set 'incognia_login_rule' as 'enabled' in your Auth0
+  // application metadata. This can be done in the advanced settings of your app.
+  const incogniaLoginRule = _.get(context, 'clientMetadata.incognia_login_rule');
+  if (!incogniaLoginRule || incogniaLoginRule !== 'enabled') {
+    console.log('Incognia login rule is not enabled for this client. Skipping');
+    return callback(null, user, context);
+  }
+
+  const installationId = _.get(context, 'request.query.installation_id');
+  if (!installationId) {
+    console.log('Missing installation_id. Skipping.');
+    return callback(null, user, context);
+  }
+
+  const accountId = _.get(user, 'user_id');
+  if (!accountId) {
+    console.log('Missing user_id. Skipping.');
+    return callback(null, user, context);
+  }
+
+  let incogniaAPI;
+  if (global.incogniaAPI) {
+    incogniaAPI = global.incogniaAPI;
+  } else {
+    incogniaAPI = new IncogniaAPI({
+      clientId: INCOGNIA_CLIENT_ID,
+      clientSecret: INCOGNIA_CLIENT_SECRET
+    });
+    global.incogniaAPI = incogniaAPI;
+  }
+
+  try {
+    const loginAssessment = await incogniaAPI.registerLoginAssessment({
+      installationId: installationId,
+      accountId: accountId
+    });
+    // TODO: should this be in context.idToken instead? This is information about the request,
```

Comment:
@joshcanhelp I'd be happy to discuss this. I saw that other marketplace integrations sometimes use `context.idToken` to fill with information about the request. 

The risk assessment is about the login itself, not about the user. Hence my doubt

Reply:
If you think the mobile app requesting login would want to know something about the outcome then I would add it or give the option to add it with a config flag, like `INCOGNIA_ADD_TO_ID_TOKEN` or thereabouts. If it's unlikely that's the case, then it can be left out.

Reply:
I think there's no need for the mobile app to get this information. Thanks! So we should stick with `user.app_metadata` only, right? Even if the risk assessment is about the login itself, and not the user?

Reply:
I guess that would depend as well, I could see it working both ways. If the risk assessment is meant to inform other applications or create a record that an admin can view then, yes, it seems like a good idea to persist. If it really is only helpful for this specific login, then maybe not.